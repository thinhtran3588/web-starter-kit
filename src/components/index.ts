export * from './AdminLayout';
export * from './AdminFooter';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Layout';
export * from './Sidebar';
export * from './SidebarMenu';
export * from './FormHeader';
export * from './Field';
export * from './FormFilter';
export * from './FormInput';
export * from './FormSearch';
export * from './Table';
export * from './LanguageSelection';
export * from './Link';
export * from './AuthLayout';
export * from './Head';
export * from './Button';
export * from './Grid';
export * from './Typography';
