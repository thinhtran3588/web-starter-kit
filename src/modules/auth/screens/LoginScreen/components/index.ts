export * from './EmailLogin';
export * from './PhoneNoLogin';
