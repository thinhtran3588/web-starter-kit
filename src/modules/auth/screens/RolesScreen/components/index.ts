export * from './Detail';
export * from './PermissionsTable';
