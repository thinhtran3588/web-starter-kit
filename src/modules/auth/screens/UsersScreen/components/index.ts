export * from './Detail';
export * from './PermissionsTable';
export * from './ChangePassword';
