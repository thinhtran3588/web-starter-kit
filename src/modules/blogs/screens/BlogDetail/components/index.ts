export * from './Content';
export * from './SideMenu';
