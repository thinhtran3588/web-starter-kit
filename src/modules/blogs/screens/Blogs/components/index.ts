export * from './Categories';
export * from './BlogsList';
