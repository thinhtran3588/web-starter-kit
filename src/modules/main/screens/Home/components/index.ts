export * from './CoverImage';
export * from './Features';
export * from './Quote';
export * from './Help';
