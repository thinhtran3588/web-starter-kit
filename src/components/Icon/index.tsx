import React from 'react';
import Home from '@material-ui/icons/Home';
import MenuBook from '@material-ui/icons/MenuBook';
import Info from '@material-ui/icons/Info';
import Security from '@material-ui/icons/Security';
import Language from '@material-ui/icons/Language';
import Person from '@material-ui/icons/Person';
import Group from '@material-ui/icons/Group';
import { SvgIconProps } from '@material-ui/core/SvgIcon';

interface Props extends SvgIconProps {
  name: string;
}

export const Icon = (props: Props): JSX.Element => {
  const { name, ...other } = props;
  switch (name) {
    case 'Home':
      return <Home {...other} />;
    case 'MenuBook':
      return <MenuBook {...other} />;
    case 'Info':
      return <Info {...other} />;
    case 'Security':
      return <Security {...other} />;
    case 'Language':
      return <Language {...other} />;
    case 'Person':
      return <Person {...other} />;
    case 'Group':
      return <Group {...other} />;
    default:
      return <></>;
  }
};
