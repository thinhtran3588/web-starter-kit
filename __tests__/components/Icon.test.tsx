import React from 'react';
import { render } from '@testing-library/react';
import { Icon } from '@app/components/Icon';

describe('components/Icon', () => {
  beforeEach(() => {});

  it('renders successfully', async () => {
    const { baseElement } = render(<Icon name='Home' />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders default successfully', async () => {
    const { baseElement } = render(<Icon name='default' />);

    expect(baseElement).toMatchSnapshot();
  });
});
