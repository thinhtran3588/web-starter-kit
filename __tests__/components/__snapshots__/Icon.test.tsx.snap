// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Icon renders default successfully 1`] = `
<body>
  <div />
</body>
`;

exports[`components/Icon renders successfully 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </div>
</body>
`;
